### PR TITLE
Implement del RPC

### PR DESCRIPTION
**Pull request by aboodman**
Here's ya dang rpc.

Manually tested the delete button in the todo sample works now!

Fixes #154

### CODE REVIEW

File: src/embed/connection.rs [hunk 1]
```diff
@@ -398,6 +399,17 @@ async fn do_put(txn: &RwLock<Transaction<'_>>, req: PutRequest) -> Result<PutRes
     Ok(PutResponse {})
 }
 
+async fn do_del(txn: &RwLock<Transaction<'_>>, req: DelRequest) -> Result<DelResponse, String> {
+    let mut guard = txn.write().await;
+    let write = match &mut *guard {
+        Transaction::Write(w) => Ok(w),
+        Transaction::Read(_) => Err("Specified transaction is read-only".to_string()),
+    }?;
+    let had = write.as_read().has(req.key.as_bytes());
```
Comment (aboodman) [inline src/embed/connection.rs:408]: Sadly using the return value from the underlying data structure is not so simple as I made it seem.

Comment (phritz) [reply src/embed/connection.rs:401]: yeah i started thinking it was going to be easy and then gave up until i had more time

File: src/embed/types.rs [hunk 2]
```diff
@@ -184,6 +184,18 @@ pub struct PutRequest {
...
 pub struct PutResponse {}
 
+#[derive(DeJson)]
+pub struct DelRequest {
+    #[nserde(rename = "transactionId")]
+    pub transaction_id: u32,
+    pub key: String,
+}
+
+#[derive(DeJson, SerJson)]
```
Comment (phritz) [inline src/embed/types.rs:194]: "ok" makes it sound like if ok is false there was a problem. how about 'was_present' or 'had' similar? at the very least we should have a comment.

**Review by phritz — APPROVED** [review]

File: src/embed/types.rs (hunk 2 above)
Comment (aboodman) [reply src/embed/types.rs:194]: Yeah it needs to be `ok` for compat with the js, but I can use rename so at least the field is reasonable.

Comment (aboodman) [reply src/embed/types.rs:194]: Fixed

**Pull request merged**